File: Server/models/transaction.py
class Transaction:
    def __init__(self, id, user_id, vendor, amount, category_name, is_deleted=False):

        self.id = id
        self.user_id = user_id
        self.vendor = vendor
        self.amount = amount
        self.category_name = category_name
        self.is_deleted = is_deleted

    def __str__(self):
        return str(self.__dict__)

    def __repr__(self) -> str:
        return self.__str__()

File: Frontend/bank/src/pages/transaction.js

import TableCell from '@mui/material/TableCell';
import TableRow from '@mui/material/TableRow';
import Button from '@mui/material/Button';
import DeleteIcon from '@mui/icons-material/Delete';
import TransactionsApi from '../data/transactionsApi';


function Transaction(props) {

    const { row } = props
    return (<TableRow
        key={row.name}
        sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
    >
        <TableCell component="th" scope="row">
            {row.name}
        </TableCell>
        <TableCell align="right">{row.vendor}</TableCell>
        <TableCell align="right">{row.amount}</TableCell>
        <TableCell align="right">{row.category_name}</TableCell>
        <TableCell align="right"><Button variant="outlined" onClick={() => props.deleteTransaction(row.id)} startIcon={<DeleteIcon />}>
            Delete
        </Button></TableCell>
    </TableRow>)
}

export default Transaction;